azurerm/internal/services/resource/parse/tenant_template_deployment.go

```diff
@@ -0,0 +1,84 @@
+package parse
+
+import (
+	"fmt"
+	"net/url"
+	"strings"
+
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+)
+
+type TenantTemplateDeploymentId struct {
+	DeploymentName string
+}
+
+func NewTenantTemplateDeploymentID(deploymentName string) TenantTemplateDeploymentId {
+	return TenantTemplateDeploymentId{
+		DeploymentName: deploymentName,
+	}
+}
+
+func (id TenantTemplateDeploymentId) String() string {
+	segments := []string{
+		fmt.Sprintf("Deployment Name %q", id.DeploymentName),
+	}
+	segmentsStr := strings.Join(segments, " / ")
+	return fmt.Sprintf("%s: (%s)", "Tenant Template Deployment", segmentsStr)
+}
+
+func (id TenantTemplateDeploymentId) ID() string {
+	fmtString := "/providers/Microsoft.Resources/deployments/%s"
```

Comment:
from memory these can also be `/tenants/0000/...` for other tenants too, so perhaps it's worth making that here?

Reply:
There's (currently) no way to specify a non-local tenant as the deployment target that I can see, so I think that would be an unnecessary complication here?